recipes/tcl/8.6.10/conanfile.py

```diff
@@ -101,72 +100,76 @@ def _build_nmake(self, targets):
-        )
-
-    def _build_autotools(self):
+        with tools.vcvars(self.settings):
+            with tools.chdir(self._get_configure_dir("win")):
+                self.run('nmake -nologo -f "{cfgdir}/makefile.vc" INSTALLDIR="{pkgdir}" OPTS={opts} {targets}'.format(
+                    cfgdir=self._get_configure_dir("win"),
+                    pkgdir=self.package_folder,
+                    opts=",".join(opts),
+                    targets=" ".join(targets),
+                ))
+
+    def _configure_autotools(self):
+        if self._autotools:
+            return self._autotools
+        self._autotools = AutoToolsBuildEnvironment(self, win_bash=tools.os_info.is_windows)
         conf_args = [
             "--enable-threads",
             "--enable-shared" if self.options.shared else "--disable-shared",
             "--enable-symbols" if self.settings.build_type == "Debug" else "--disable-symbols",
             "--enable-64bit" if self.settings.arch == "x86_64" else "--disable-64bit",
-            "--prefix={}".format(self.package_folder),
         ]
-        autoTools = self._get_auto_tools()
-        autoTools.configure(configure_dir=self._get_configure_dir(), args=conf_args, vars={"PKG_CFG_ARGS": " ".join(conf_args)})
+        self._autotools.configure(configure_dir=self._get_configure_dir(), args=conf_args, vars={"PKG_CFG_ARGS": " ".join(conf_args)})
 
         # https://core.tcl.tk/tcl/tktview/840660e5a1
         for root, _, files in os.walk(self.build_folder):
             if "Makefile" in files:
                 tools.replace_in_file(os.path.join(root, "Makefile"), "-Dstrtod=fixstrtod", "", strict=False)
-
-        with tools.chdir(self.build_folder):
-            autoTools.make()
+        return self._autotools
 
     def build(self):
-        self._fix_sources()
+        self._patch_sources()
         if self.settings.compiler == "Visual Studio":
-            # do not treat warnings as errors 
-            tools.replace_in_file(os.path.join(self._source_subfolder, "win", "rules.vc"), "cwarn = $(cwarn) -WX", "")
             self._build_nmake(["release"])
         else:
-            self._build_autotools()
+            autotools = self._configure_autotools()
+            autotools.make()
 
     def package(self):
+        self.copy(pattern="license.terms", dst="licenses", src=self._source_subfolder)
         if self.settings.compiler == "Visual Studio":
             self._build_nmake(["install-binaries", "install-libraries"])
         else:
-            with tools.chdir(self.build_folder):
-                autoTools = self._get_auto_tools()
+            autoTools = self._configure_autotools()
+            if self.settings.os == "Windows":
+                # install-headers target is not available on Windows, so do a plain install
+                autoTools.install()
+                autoTools.make(target="install-private-headers")
+            else:
                 autoTools.make(target="install-binaries")
```

Comment:
@theodelrieu Why do you do `make install-binaries`, `make install-libraries`, `make install-msgs`, ... instead of a single `make install`?

Reply:
I don't remember exactly what I've modified vs. what was on the original recipe, I would go with the single `make install` and repackage everything indeed!

Reply:
Thanks, then I'll just do a single `make install` for all.